rules/aip0133/resource_reference_type_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0133
+
+import (
+	"testing"
+
+	"github.com/googleapis/api-linter/rules/internal/testutils"
+)
+
+func TestResourceReferenceType(t *testing.T) {
+	// Set up testing permutations.
+	tests := []struct {
+		testName string
+		TypeName string
+		RefType  string
+		Response string
+		problems testutils.Problems
+	}{
+		{"ValidChildType", "library.googleapis.com/Book", "child_type", "Book", nil},
+		{"ValidChildTypeLRO", "library.googleapis.com/Book", "child_type", "google.longrunning.Operation", nil},
+		{"ValidType", "library.googleapis.com/Shelf", "type", "Book", nil},
+		{"InvalidType", "library.googleapis.com/Book", "type", "Book", testutils.Problems{{Message: "not a type"}}},
+		{"InvalidChildType", "library.googleapis.com/Shelf", "child_type", "Book", testutils.Problems{{Message: "child_type"}}},
+	}
+
+	// Run each test.
+	for _, test := range tests {
+		t.Run(test.testName, func(t *testing.T) {
+			file := testutils.ParseProto3Tmpl(t, `
+				import "google/api/resource.proto";
+				import "google/longrunning/operations.proto";
+				service Library {
+					rpc CreateBook(CreateBookRequest) returns ({{ .Response }}) {
+						option (google.longrunning.operation_info) = {
+							response_type: "Book"
+							metadata_type: "Book"
+						};
```

Comment:
It seems odd to include `option`, `google.longrunning.operation_info`, for the cases that are not LROs. I would suggest doing one of the following

1. Have two different top level tests with different test case arrays, one per top level test, and different top level test method bodies for Operations and without Operations.
2. Have one top level test case with two different test case arrays, in the same top level test, with different loops and test bodies, within each loop.
3. In the template, use a conditional that omits the `operation_info` iff the `.Response` does not `Equals` `google.longrunning.Operations`

If you opt for 1 or 2, above, you can dry off with a private method, optionally.

Reply:
Went with option 1 :)